File: public/dsa/searching/non_repeating_element/naive/non_repeating_element.py
from typing import List


class Solution:
    def findNonRepeatingElement(self, arr: List[int]) -> int:
        n = len(arr)
        for i in range(0, n - 1, 2):
            if arr[i] != arr[i + 1]:
                return arr[i]
        return arr[n - 1]


# tests-start
arr = [1, 1, 2, 3, 3, 4, 4]
sol = Solution()
print(sol.findNonRepeatingElement(arr))
# tests-end

'''output
2
'''

File: public/dsa/searching/non_repeating_element/naive/non_repeating_element.cpp

#include <bits/stdc++.h>
using namespace std;

int findNonRepeatingElement(vector<int> &arr) {
    int n = arr.size();
    for (int i = 0; i < n - 1; i += 2) {
        if (arr[i] != arr[i + 1]) {
            return arr[i];
        }
    }
    return arr[n - 1];
}

// tests-start
int main() {
    vector<int> arr = {1, 1, 2, 3, 3, 4, 4};
    cout << findNonRepeatingElement(arr) << endl;
    return 0;
}
// tests-end

/*output
2
*/